docs/proposals/daemon-controller.md

```diff
@@ -0,0 +1,151 @@
+
+In Kubernetes, a Replication Controller ensures that the specified number of a specified pod are running in the cluster at all times (pods are restarted if they are killed). With the Replication Controller, users cannot control which nodes their pods run on - Kubernetes decides how to schedule the pods onto nodes. However, many users want control over how certain pods are scheduled. In particular, many users have requested for a way to run a daemon on every node in the cluster, or on a certain set of nodes in the cluster. This is essential for use cases such as building a sharded datastore, or running a logger on every node. In comes the daemon controller, a way to conveniently create and manage daemon-like workloads in Kubernetes.  
+
+## Use Cases
+
+The daemon controller can be used for user-specified system services, cluster level applications with strong node ties, and Kubernetes node services. Below are example use cases in each category.
+
+### User-Specified System Services:
+Logging: Some users want a way to collect statistics about nodes in a cluster and send those logs to an external database. For example, system administrators might want to know if their machines are performing as expected, if they need to add more machines to the cluster, or if they should switch cloud providers. The daemon controller can be used to run a data collection service (for example fluentd) and send the data to a service like ElasticSearch for analysis.
+
+### Cluster-Level Applications
+Datastore: Users might want to implement a sharded datastore in their cluster. A few nodes in the cluster, labeled ‘datastore’, might be responsible for storing data shards, and pods running on these nodes might serve data. This architecture requires a way to bind pods to specific nodes, so it cannot be achieved using a Replication Controller. A daemon controller is a convenient way to implement such a datastore. At Google, D is an example of such an application.
+
+For other uses, see the related [feature request](https://github.com/GoogleCloudPlatform/kubernetes/issues/1518)
+
+## Functionality
+
+The Daemon Controller will support standard API features:
+- create
+  - The spec for daemon controllers will have a pod template field.  Note that daemon controllers will not have a nodename field.
+  - Using the pod’s node selector field, Daemon controllers can be restricted to operate over nodes that have a certain label. For example, suppose that in a cluster some nodes are labeled ‘database’. You can use a daemon controller to launch a datastore pod on exactly those nodes labeled ‘database’.
+  - The spec for pod templates that run with the Daemon Controller is the same as the spec for pod templates that run with the Replication Controller, except there will not be a ‘replicas’ field (exactly 1 daemon will be launched per node).
+  - We will not guarantee that daemon pods show up on nodes before regular pods - run ordering is out of scope for this controller
```

Comment:
I don't think that would be too hard, actually, esp. if this functionality were added to the node controller, as discussed in #1518. In any case, I don't think we should preclude it, though it doesn't need to happen in the initial implementation.